x-pack/platform/plugins/shared/task_manager/server/lib/api_key_utils.ts

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { SecurityServiceStart } from '@kbn/core/server';
+import { KibanaRequest } from '@kbn/core/server';
+import { SpacesPluginStart } from '@kbn/spaces-plugin/server';
+import { truncate } from 'lodash';
+import { TaskUserScope } from '../task';
+
+export interface GrantAPIKeyResult {
+  id: string;
+  name: string;
+  api_key: string;
+}
+
+const getCredentialsFromRequest = (request: KibanaRequest) => {
+  const authorizationHeaderValue = request.headers.authorization;
+  if (!authorizationHeaderValue || typeof authorizationHeaderValue !== 'string') {
+    return null;
+  }
+  const [scheme] = authorizationHeaderValue.split(/\s+/);
+  return authorizationHeaderValue.substring(scheme.length + 1);
+};
+
+export const isRequestApiKeyType = (request: KibanaRequest, security: SecurityServiceStart) => {
+  const user = security.authc.getCurrentUser(request);
+  return user?.authentication_type === 'api_key';
+};
+
+export const getApiKeyFromRequest = (request: KibanaRequest, name: string) => {
+  const credentials = getCredentialsFromRequest(request);
+  if (credentials) {
+    const apiKey = Buffer.from(credentials, 'base64').toString().split(':');
+
+    return {
+      name,
+      id: apiKey[0],
+      api_key: apiKey[1],
+    };
+  }
+  return null;
+};
+
+export const createApiKey = async (
+  request: KibanaRequest,
+  canEncryptSo: boolean,
+  security: SecurityServiceStart
+) => {
+  if (!canEncryptSo) {
+    throw Error(
+      'Unable to create API keys because the Encrypted Saved Objects plugin has not been registered or is missing encryption key.'
+    );
+  }
+
+  if (!(await security.authc.apiKeys.areAPIKeysEnabled())) {
```

Comment:
seems like something we could check when the task store class is initialized and set as a boolean so we don't have to check every time? it shouldn't change.